### DISCUSSION TITLE
TH16 alternative?

### DISCUSSION THREAD

**Discussion by treb42** [BernhardSchlegel/BierBot-Bricks#32] [General]
the TH16 seems to be pretty hard to get at the moment. How are chances to get the software running on an esp32 like the Sonoff TH Origin  (THR320 or THR320D)?

Thanks,
Markus

**Comment by BernhardSchlegel** [comment]
Hey Markus, thanks for reaching out. Sonoff Origin is literally on my table. It features a more powerful ESP32 and 3LEDs which requires some modifications (or enables optimizations as the ESP32 features 2 cores).

The Firmware will be released (open sourced) shortly, maybe even as early as next week.

The TH16 is EOP and the Origin is it's official successor.

Cheers,
Bernhard.

**Reply by treb42** [reply]
great! and take your time :) aliexpress ain't that fast shipping mine!

**Comment by BernhardSchlegel** [comment]
Hey Markus (@treb42),

the code for Sonoff TH Origin is online. Still need to clean it up a bit, but basically everything works. See here: https://github.com/BernhardSchlegel/BierBot-Brick-ESP32 - but in general you have to compile the code yourself with VSCode and Platform.io (no release yet).

Instructions and video will come in the next days!

Cheers,
Bernhard.
